docs/layers/terrain-layer.md

```diff
@@ -0,0 +1,137 @@
+<!-- INJECT:"TerrainLayerDemo" -->
+
+<p class="badges">
+  <img src="https://img.shields.io/badge/@deck.gl/geo--layers-lightgrey.svg?style=flat-square" alt="@deck.gl/geo-layers" />
+  <img src="https://img.shields.io/badge/lighting-yes-blue.svg?style=flat-square" alt="lighting" />
+</p>
+
+# TerrainLayer
+
+This TerrainLayer reconstructs mesh surfaces from height map images, e.g. [Mapzen Terrain Tiles](https://github.com/tilezen/joerd/blob/master/docs/formats.md), which encodes elevation into R,G,B values.
+
+When `terrainImage` is supplied with a templated tile server URL, e.g. URL containing "{x}", it renders terrain tiles globally using the TileLayer and SimpleMeshLayer. Otherwise, supply an absolute URL to `terrainImage` and a `bounds` prop to render a any terrain image (e.g. non-Slippy map tiles) with a SimpleMeshLayer.
+
+```js
+import DeckGL from '@deck.gl/react';
+import {TerrainLayer} from '@deck.gl/geo-layers';
+
+export const App = ({viewport}) => {
+
+  const layer = new TerrainLayer({
+    minZoom: 0,
+    maxZoom: 23,
+    strategy: 'no-overlap',
+    elevationDecoder: {
+      rScaler: 256,
+      gScaler: 1,
+      bScaler: 1/256,
+      offset: -32768
+    },
+    terrainImage: 'https://s3.amazonaws.com/elevation-tiles-prod/terrarium/{z}/{x}/{y}.png',
+    surfaceImage: 'https://wms.chartbundle.com/tms/1.0.0/sec/{z}/{x}/{y}.png?origin=nw'
+  });
+  return <DeckGL {...viewport} layers={[layer]} />;
+};
+```
+
+## Installation
+
+To install the dependencies from NPM:
+
+```bash
+npm install deck.gl
+# or
+npm install @deck.gl/core @deck.gl/layers @deck.gl/geo-layers
+```
+
+```js
+import {TerrainLayer} from '@deck.gl/geo-layers';
+new TerrainLayer({});
+```
+
+To use pre-bundled scripts:
+
+```html
+<script src="https://unpkg.com/deck.gl@^8.0.0/dist.min.js"></script>
+<!-- or -->
+<script src="https://unpkg.com/@deck.gl/core@^8.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/@deck.gl/layers@^8.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/@deck.gl/geo-layers@^8.0.0/dist.min.js"></script>
+```
+
+```js
+new deck.TerrainLayer({});
+```
+
+## Properties
+
+When in Tiled Mode, inherits from all [Tile Layer](/docs/api-reference/tile-layer.md) properties. Otherwise, inherits from all [Simple Mesh Layer](/docs/api-reference/simple-mesh-layer.md) properties.
```

Comment:
> forwarded

Is this just `bounds`?

Reply:
We usually avoid blindly pass `this.props` to a sublayer. In case the sublayer may add new props that conflict with the parent layer's functionalities. An example is the [GeoJSONLayer](https://github.com/uber/deck.gl/blob/master/modules/layers/src/geojson-layer/geojson-layer.js#L161), which only explicitly forward props. 

With that said, I think what you're doing in this layer is fine. In the documentation, you should always explicitly document the props that are explicitly forwarded (looks like it's only the `wireframe` prop of `SimpleMeshLayer`), and only say "inherits from..." if all props are implicitly forwarded (props of `TileLayer`). You are not handling any other `SimpleMeshLayer` props such as `getOrientation`, `getTransformMatrix`, `sizeScale` etc. so it's inaccurate to claim that this layer inherits all of its properties.

Reply:
Ok I took a crack at rephrasing this.